zerver/lib/bugdown/__init__.py

```diff
@@ -540,11 +540,18 @@ def youtube_id(self, url: Text) -> Optional[Text]:
             return None
         return match.group(2)
 
-    def youtube_image(self, url: Text) -> Optional[Text]:
-        yt_id = self.youtube_id(url)
+    def youtube_image(self, extracted_data: Dict[Text, Any]) -> Optional[Text]:
+        yt_image = extracted_data.get('image')
 
-        if yt_id is not None:
-            return "https://i.ytimg.com/vi/%s/default.jpg" % (yt_id,)
+        if yt_image is not None:
```

Comment:
Stylistically, I would rewrite this code as follows:

~~~
if yt_image is None:
    # not all videos have images, so exiting early is fine
    return None

return yt_image
~~~

But of course you can see that the above code really boils down to `return yt_image`.   If using the extracted data prevents us from having to do the string formatting, this code should become really simple.